core/src/main/java/tech/tablesaw/api/Table.java

```diff
@@ -454,6 +454,28 @@ public Table emptyCopy(int rowSize) {
         tables[1] = where(table2Selection);
         return tables;
     }
+    
+    /**
+     * Splits the table into two stratified samples, this uses the specified column to divide the table into groups, randomly assigning records to each according to the proportion given in trainingProportion. 
+     * @param column the column to be used for the stratified sampling
+     * @param table1Proportion The proportion to go in the first table
+     * @return An array two tables, with the first table having the proportion specified in the method parameter,
+     * and the second table having the balance of the rows
+     */
+    public Table[] stratifiedSampleSplit(CategoricalColumn column, double table1Proportion) {
+        Preconditions.checkArgument(containsColumn(column),
+                "The categorical column must be part of the table, you can create a string column and add it to this table before sampling.");
+        final Table first = emptyCopy();
```

Comment:
I think you could rewrite this method as:

```
        return splitOn(column).asTableList().stream().map(t -> t.sampleSplit(table1Proportion)).reduce(new Table[2], (acc, res) -> {
            acc[0].append(res[0]);
            acc[1].append(res[1]);
            return acc;
        });
```

Reply:
I will take a look at it this week, sorry for the delay